### PR TITLE
Add docker operations and docker volume facts suport

### PR DESCRIPTION
**Pull request by apecnascimento**
This PR Add support  to:

Facts:
- Docker volume
- Docker Volumes

Operations:
- Docker containers
- Docker images
- Docker networks
- Docker volumes
- Docker prune

### CODE REVIEW

**Review by alessonviana — APPROVED** [review]

**Comment by Fizzadar** [comment]
Well this is just incredible @apecnascimento! Thank you very much for working on this, probably the most requested operation/fact expansion in pyinfra history! Will do a proper review this week 🙏

File: pyinfra/facts/util/docker.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import json
```
Comment (Fizzadar) [inline pyinfra/facts/util/docker.py]: This file appears to be entirely unused?

File: pyinfra/operations/docker.py [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+from pyinfra.facts.docker import DockerContainers, DockerNetworks, DockerVolumes
+
+from .util.docker import handle_docker
+
+
+@operation()
+def container(
+    container,
+    image="",
+    ports=[],
```
Comment (Fizzadar) [inline pyinfra/operations/docker.py]: Mutable arguments are dangerous in Python, we should default to `None` for these and default create the empty lists within the function body.

File: pyinfra/operations/docker.py [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+    else:
+        yield handle_docker(
+            resource="image",
+            command="remove",
+            image=image,
+        )
+
+
+@operation()
+def volume(volume, driver="", labels=[], present=True):
```
Comment (Fizzadar) [inline pyinfra/operations/docker.py]: Default list here

File: pyinfra/operations/docker.py [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+@operation()
+def network(
+    network,
+    driver="",
+    gateway="",
+    ip_range="",
+    ipam_driver="",
+    subnet="",
+    scope="",
+    opts=[],
```
Comment (Fizzadar) [inline pyinfra/operations/docker.py]: Default lists here

**Review by Fizzadar — CHANGES_REQUESTED** [review]
Sorry for taking so long to get to this, @apecnascimento. Looks great overall, couple of issues with default arguments that need cleaning up first then good to go :)

File: pyinfra/facts/util/docker.py (hunk 1 above)
Comment (apecnascimento) [reply pyinfra/facts/util/docker.py]: resolved in commit [c961154](https://github.com/pyinfra-dev/pyinfra/pull/1114/commits/c961154d198a89f396ba654a2cadadaf4e424368)

File: pyinfra/operations/docker.py (hunk 2 above)
Comment (apecnascimento) [reply pyinfra/operations/docker.py]: resolved in commit [0ea1976](https://github.com/pyinfra-dev/pyinfra/pull/1114/commits/0ea1976f4fa2881d636e952a0f2018e891d47434)

File: pyinfra/operations/docker.py (hunk 3 above)
Comment (apecnascimento) [reply pyinfra/operations/docker.py]: resolved in commit [0ea1976](https://github.com/pyinfra-dev/pyinfra/pull/1114/commits/0ea1976f4fa2881d636e952a0f2018e891d47434)

File: pyinfra/operations/docker.py (hunk 4 above)
Comment (apecnascimento) [reply pyinfra/operations/docker.py]: resolved in commit [0ea1976](https://github.com/pyinfra-dev/pyinfra/pull/1114/commits/0ea1976f4fa2881d636e952a0f2018e891d47434)

**Review by Fizzadar — APPROVED** [review]
Awesome stuff @apecnascimento! Thank you for making those changes, super excited for this to land in pyinfra v3!

**Comment by apecnascimento** [comment]
Tks @Fizzadar  for your hard work. I can imagine how hard it's to check all those PRs. I'm really excited for v3.

**Pull request merged**